run/src/main/kotlin/com/cosmotech/run/workflow/argo/RunArgoWorkflowService.kt

```diff
@@ -226,25 +224,55 @@ internal class RunArgoWorkflowService(
     return buildRunStatusFromWorkflowStatus(run, workflowStatus)
   }
 
-  override fun getRunLogs(run: Run): RunLogs {
+  override fun getRunningLogs(run: Run): String {
     val workflowName = run.workflowName
-    var containersLogs = listOf<RunLogsEntry>()
+    var result = ""
+    val clientBuilder = RestClient.builder()
```

Comment:
You can define the RestClient as a class property instead.

Something like this:

`private val argoClient = RestClient.builder().baseUrl(csmPlatformProperties.argo.baseUri).build()`